block/slvalidator.go

```diff
@@ -172,36 +170,29 @@ func (v *SettlementValidator) ValidateDaBlocks(slBatch *settlement.ResultRetriev
 		if err != nil {
 			return err
 		}
+	}
 
-		// we compare the sequencer address between SL state info and DA block
-		// if next sequencer is not set, we check if the sequencer hash is equal to the next sequencer hash
-		// because it did not change. If the next sequencer is set, we check if the next sequencer hash is equal on the
-		// last block of the batch
-		isLastBlock := i == len(slBatch.BlockDescriptors)-1
-		if slBatch.NextSequencer != currentProposer.SettlementAddress && isLastBlock {
-			err := v.blockManager.UpdateSequencerSetFromSL()
-			if err != nil {
-				return fmt.Errorf("update sequencer set from SL: %w", err)
-			}
-			nextSequencer, found := v.blockManager.Sequencers.GetByAddress(slBatch.NextSequencer)
-			if !found {
-				return fmt.Errorf("next sequencer not found")
-			}
-			if !bytes.Equal(nextSequencer.MustHash(), daBlocks[i].Header.NextSequencersHash[:]) {
-				return types.NewErrInvalidNextSequencersHashFraud(
-					[32]byte(nextSequencer.MustHash()),
-					daBlocks[i].Header.NextSequencersHash,
-				)
-			}
-		} else {
-			if !bytes.Equal(daBlocks[i].Header.SequencerHash[:], daBlocks[i].Header.NextSequencersHash[:]) {
-				return types.NewErrInvalidNextSequencersHashFraud(
-					daBlocks[i].Header.SequencerHash,
-					daBlocks[i].Header.NextSequencersHash,
-				)
-			}
+	// we compare the sequencer address between SL state info and DA block
+	// if next sequencer is not set, we check if the sequencer hash is equal to the next sequencer hash
+	// because it did not change. If the next sequencer is set, we check if the next sequencer hash is equal on the
+	// last block of the batch
+	lastDABlock := daBlocks[len(daBlocks)-1]
+	expectedNextSeqHash := lastDABlock.Header.SequencerHash
+	if slBatch.NextSequencer != slBatch.Sequencer {
```

Comment:
1. instead of using state proposer, doing `slBatch.NextSequencer != slBatch.Sequencer` to check for a change
2. skipped `UpdateSequencerSetFromSL` as we can assume sync with SL metadata